clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -1008,9 +974,9 @@ private Fetch<K, V> pollForFetches(Timer timer) {
      * @throws org.apache.kafka.common.errors.AuthenticationException if authentication fails. See the exception for more details
      * @throws NoOffsetForPartitionException If no offset is stored for a given partition and no offset reset policy is
      *             defined
-     * @return true iff the operation completed without timing out
+     * @return true if the operation completed without timing out
      */
-    private boolean updateFetchPositions(final Timer timer) {
+    private boolean updateFetchPositions() {
         // If any partitions have been truncated due to a leader change, we need to validate the offsets
         ResetPositionsApplicationEvent event = new ResetPositionsApplicationEvent();
         eventHandler.add(event);
```

Comment:
I feel the logic needs to be refactored a bit here.  As in the current implementation, `        subscriptions.resetInitializingPositions()` is a blocked by refreshing the committed offset, which issues a `PendingCommittedOffsetRequest`. I'm also not sure why are we sending two reset position application event.